typings/connection.d.ts

```diff
@@ -0,0 +1,544 @@
+/// <reference types="node" />
```

Comment:
@grs - Files from here onwards are the typescript type definitions. I have kept matching file names inside the typings folder with the filenames in the lib folder. This should make it easy to review/compare the type definitions to the source code in corresponding `.js` files.

I am sure with your input the type definitions can be more accurate as you have better understanding of what those objects, methods are.